packages/core/src/signals/SignalContext.ts

```diff
@@ -101,21 +110,26 @@ export class SignalContext<
     this.markDirty();
   }
 
-  public set(value: SignalValue<TSetterValue>): TOwner {
+  public set(value: SignalValue<TSetterValue> | typeof DEFAULT): TOwner {
+    if (value === DEFAULT) {
+      value = this.initial!;
+    }
+    const resolvedValue = value as SignalValue<TSetterValue>;
```

Comment:
With `typeof DEFAULT` this is no longer necessary bc typescript will do it's magic type narrowing.